### PR TITLE
Remove private_key qesap terraform variables

### PR DESCRIPTION
**Pull request by mpagot**
Remove the Terraform variable about the private ssh key from any config.yaml. In particular remove it from any GCP

- Related ticket: https://jira.suse.com/browse/TEAM-9123

:warning:  [PR/218](https://github.com/SUSE/qe-sap-deployment/pull/218) can be merged whenever we like, but first. Even if merged it does not require any change on the openQA side.  This PR has to be merged as second and require we create a qe-sap-deployment release and change some JobGroups to use it

# Verification run:
Tested with https://github.com/SUSE/qe-sap-deployment/pull/218

### GCP
 - sle-15-SP5-Qesap-Gcp-Byos-x86_64-BuildLATEST_GCE_SLE15_5_BYOS-qesap_gcp_sapconf_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277987 :green_circle: 

The Terraform warning is gone http://openqaworker15.qa.suse.cz/tests/277987/logfile?filename=deploy-qesap_exec_terraform.log.txt . It is expected as this VR also have openQA side code to remove `private_key` from the conf.yaml

 http://openqaworker15.qa.suse.cz/tests/277992

### Azure
 - sle-15-SP5-Qesap-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_5_BYOS-qesap_azure_sapconf_test@64bit -> http://openqaworker15.qa.suse.cz/tests/277993

### Azure
 - sle-15-SP6-Azure-SAP-BYOS-x86_64-Build0467-sles4sap_gnome_saptune_delete_rename@az_Standard_E8s_v3 -> http://openqaworker15.qa.suse.cz/tests/277996

### GCP
 - sle-15-SP6-GCE-SAP-BYOS-x86_64-Build0467-sles4sap_gnome_saptune_delete_rename@gce_n1_highmem_8 -> http://openqaworker15.qa.suse.cz/tests/277986 :green_circle:  Terraform is fine. both jobs fails later for registration error.

http://openqaworker15.qa.suse.cz/tests/277995

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

File: data/sles4sap/qe_sap_deployment/mr_test_gcp.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ terraform:
     deployment_name: '%QESAP_DEPLOYMENT_NAME%'
     admin_user: 'cloudadmin'
     public_key: '%SLES4SAP_SSHKEY%.pub'
+    private_key: '%SLES4SAP_SSHKEY%'
```
Comment (alvarocarvajald) [inline data/sles4sap/qe_sap_deployment/mr_test_gcp.yaml:19]: Adding it back?

File: data/sles4sap/qe_sap_deployment/qesap_gcp.yaml [hunk 2]
```diff
@@ -12,7 +12,6 @@ terraform:
     region: '%REGION%'
     deployment_name: '%DEPLOYMENTNAME%'
     os_image: '%OS_VER%'
-    private_key: '%SSH_KEY_PRIV%'
     public_key: '%SSH_KEY_PUB%'
```
Comment (alvarocarvajald) [inline data/sles4sap/qe_sap_deployment/qesap_gcp.yaml:15]: Is it OK that in this file and the ones below it's `%SSH_KEY_PUB%` and in the file on top it's `%SLES4SAP_SSHKEY%`? Looking at the code in Perl it looks like `%SSH_KEY_PUB%` is the "correct" one.

File: data/sles4sap/qe_sap_deployment/mr_test_gcp.yaml (hunk 1 above)
Comment (mpagot) [reply data/sles4sap/qe_sap_deployment/mr_test_gcp.yaml]: wrong rebase, fixed now

File: data/sles4sap/qe_sap_deployment/qesap_gcp.yaml (hunk 2 above)
Comment (mpagot) [reply data/sles4sap/qe_sap_deployment/qesap_gcp.yaml]: yes was ok from a functional point of view (like conf.yaml for Hanasr/mr_test and for qesap_regressio are done using two different mechanism) but not ok from the point of view of code complexity. I decided to change more code to harmonize the way we compose the pub sshkey name.

More VR are running

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**